zipkin-cassandra/src/test/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStoreSpec.scala

```diff
@@ -36,4 +40,23 @@ class CassandraSpanStoreSpec extends SpanStoreSpec {
   override lazy val store = new CassandraSpanStore(new Repository(keyspace, cluster))
 
   override def clear = cluster.connect().execute("DROP KEYSPACE IF EXISTS " + keyspace)
+
+  override def setTimeToLive() {
+    ready(store(Seq(span1)))
+    ready(store.setTimeToLive(span1.traceId, 1234.seconds))
+
+    for( i <- 1 to 100) {
+      // @xxx Repository.storeXXX() methods write asynchronously but don't return a Future so we can't reliably test.
```

Comment:
That sounds pretty bad; I'd expect it can lead to all kinds of inconsistencies because the Cassandra store is not telling the truth ("truth" as defined by the API of `SpanStore`). I don't know how to formulate this properly, but maybe it's worth opening an issue to track this?